### PR TITLE
Implement negative two-phase flash

### PR DESCRIPTION
**Pull request by dkachuma**
Adds a utility to perform the iterative EOS based negative two-phase flash.

### CODE REVIEW

File: src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash_impl.hpp [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+      }
+    }
+
+    if( converged )
+    {
+      break;
+    }
+
+    // Update K-values
+    for( auto ic : presentComponentIds )
```
Comment (francoishamon) [inline src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash_impl.hpp:151]:
```suggestion
    for( integer const ic : presentComponentIds )
```
to conform to the style currently used in the code

File: src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+#include "common/DataTypes.hpp"
+
+namespace geos
+{
+
+namespace constitutive
+{
+
+template< typename EOS_TYPE_LIQUID, typename EOS_TYPE_VAPOUR >
```
Comment (francoishamon) [inline src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp:30]: Nice, we can see later if it is useful to have both templates or if having only one is enough

**Review by francoishamon — APPROVED** [review]
Looks great!

File: src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+{
+public:
+  /// Max number of components alloweeed in the class for now
+  static constexpr integer maxNumComps = 5;
+  /// Max number of iterations
+  static constexpr integer maxIterations = 200;
+  /// Epsilon used in the calculations
+  static constexpr real64 epsilon = LvArray::NumericLimits< real64 >::epsilon;
+  /// Tolerance for checking fugacity ratio convergence
+  static constexpr real64 fugacityTolerance = 1.0e-8;
```
[the remark this answers is not in the document]
Comment (dkachuma) [reply src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp:41]: Agree. I think I will do immediately after this, a follow up PR for the constants. At least to localise them in one place.

**Comment by paveltomin** [comment]
Sorry, @dkachuma, I just saw it today that you picked me as reviewer, I totally missed that.

**Comment by dkachuma** [comment]
> Sorry, @dkachuma, I just saw it today that you picked me as reviewer, I totally missed that.

No problem. Please take a look and point out anything you notice.

File: src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp (hunk 2 above)
Comment (paveltomin) [reply src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash.hpp:30]: My small suggestion here is to move the template definition down to compute function only?

**Pull request merged**